src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -559,13 +560,36 @@ private static Details getDetails(DetailAST ast) {
                 case TokenTypes.LITERAL_DO:
                     details = getDetailsForDoLoops(ast);
                     break;
+                case TokenTypes.LITERAL_SWITCH:
+                    details = getDetailsForSwitch(ast);
+                    break;
                 default:
                     details = getDetailsForOthers(ast);
                     break;
             }
             return details;
         }
 
+        /**
+         * Collect all the requirement to check violation for switch statements and expressions.
+         *
+         * @param ast Node which pointing to Token Switch case
+         * @return object containing all details to make a validation
+         */
+        private static Details getDetailsForSwitch(DetailAST ast) {
+            final DetailAST lcurly = ast.findFirstToken(TokenTypes.LCURLY);
+            final DetailAST rcurly = ast.getLastChild();
+            final DetailAST nextToken;
+            if (ast.getParent().getType() == TokenTypes.EXPR) {
```

Comment:
Can we extract condition to some descriptively named variable? Does `isSwitchExpression` make sense?

Reply:
done